actions/free-disk-space/action.yml

```diff
@@ -0,0 +1,41 @@
+name: free-disk-space
+description: "Delete specific directories or files to free disk space"
+
+inputs: {}
+
+runs:
+  using: composite
+  steps:
+    - name: Free Disk Space
+      shell: bash
+      run: |
+        # Define a list of directories or files to delete
+        TO_DELETE=(
+          "/usr/share/dotnet"
+          "/usr/local/lib/android"
+          "/usr/local/share/powershell"
+          "/usr/local/share/chromium"
+          "/usr/local/lib/heroku"
+          "/usr/local/share/boost"
+          "/usr/local/share/miniconda"
+          "/opt/hostedtoolcache/CodeQL"
+          "/opt/hostedtoolcache/Ruby"
+          "/usr/lib/jvm"
+          "/opt/hostedtoolcache/Java_*"
+          "/opt/microsoft"
+          "/opt/google"
+        )
+
+        echo "Starting disk cleanup..."
+
+        # Loop through and delete each item if it exists
+        for path in "${TO_DELETE[@]}"; do
+          if [ -e "$path" ]; then
+            echo "::info::Deleting: $path"
+            sudo rm -rf "$path" || true
```

Comment:
If this ends up being slow or as an optimization, we should consider gnu `parallel` since it's [already installed](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#installed-apt-packages).

Reply:
Added concurrency to this - went from about 55 seconds -> 43 seconds.

Not sure of the total impact, because I also reduced the number of things to delete, as I think we were clearing enough disk space.